lib/src/main/java/graphql/nadel/engine/blueprint/IntrospectionService.kt

```diff
@@ -25,22 +50,266 @@ fun interface NadelIntrospectionRunnerFactory {
+                        copy[makeFieldCoordinates(schema.queryType.name, "__type")] =
+                            Introspection.TypeMetaFieldDefDataFetcher
+                        copy[makeFieldCoordinates(schema.queryType.name, "__typename")] =
+                            GraphQLIntrospectionDataFetcher<Any> {
+                                schema.queryType.name
+                            }
+
+                        also {
+                            val coords = makeFieldCoordinates("__Type", "kind")
+                            val defaultImpl = copy[coords]
+                                ?: throw IllegalStateException("No __Type.kind data fetcher")
+                            copy[coords] =
+                                GraphQLIntrospectionDataFetcher<Any> { env ->
+                                    defaultImpl.get(env)?.toString()
+                                }
+                        }
+
+                        also {
+                            val coords = makeFieldCoordinates("__Directive", "locations")
+                            val defaultImpl = copy[coords]
+                                ?: throw IllegalStateException("No __Directive.locations data fetcher")
+                            copy[coords] =
+                                GraphQLIntrospectionDataFetcher<Any> { env ->
+                                    defaultImpl.get(env)
+                                        .let { value ->
+                                            if (value is AnyList) {
+                                                value.map {
+                                                    it.toString()
+                                                }
+                                            } else {
+                                                value
+                                            }
+                                        }
+                                }
+                        }
+                    }
+            }
+    }
+
+    private val propertyDfs = mutableMapOf<FieldCoordinates, DataFetcher<Any?>>()
 
     override fun execute(serviceExecutionParameters: ServiceExecutionParameters): CompletableFuture<ServiceExecutionResult> {
-        return graphQL
-            .executeAsync(
-                ExecutionInput.newExecutionInput()
-                    .query(AstPrinter.printAstCompact(serviceExecutionParameters.query))
-                    .variables(serviceExecutionParameters.variables)
-                    .build()
+        return CompletableFuture.completedFuture(
+            gottaGoFast(serviceExecutionParameters.executableNormalizedField)
+        )
+    }
+
+    private fun gottaGoFast(
+        field: ExecutableNormalizedField,
+    ): ServiceExecutionResult {
+        val data: MutableJsonMap = LinkedHashMap()
+        data[field.name] = getValue(field, schema)
+        return newServiceExecutionResult(data = data)
```

Comment:
So the speed increase here seems to be that we avoid parsing / validating the query again.

This is a fair saving I guess but its likely to be 10-15ms tops.  The introspection calls seems to be taking 180ms in staging.

So 165-170ms still feeds like a long time right?  I guess all savings count.

I think I would rather see this code in graphql-java

That is the ability to run introspection starting from a ENF say.

That way we can set up the code without all the relfection malarky